### DISCUSSION TITLE
How to add new block dynamically using selected component with all its styles?

### DISCUSSION THREAD

**Discussion by FaisalShaikhHA** [GrapesJS/grapesjs#4853] [General]
How to add new blocks dynamically using a selected component with all its styles?
Same as copy command does, it copies the component I want to create a block using those details.
```
editor.Blocks.add(id, {
     label: name,
     category: "Custom",
     attributes: { class: "fa fa-box" },
     content: Json.Parse(Json.stringfy(editor.getSelected().clone())),
     activate: true,
});
```
The above code adds the block but the cloned component does not include styles applied to it or CSS rules.

**Comment by allenhwkim** [comment]
AFAIK, attributes you added there does not apply to block in iframe view. 
Those attributes applies to each block in block manager.

<img width="478" alt="image" src="https://user-images.githubusercontent.com/1437734/213595421-81ee60d4-c300-44f2-afaf-9b03da5d304c.png">

<img width="302" alt="image" src="https://user-images.githubusercontent.com/1437734/213595380-8e2449a4-8d01-4e7e-ab4b-73dcc2776101.png">

**Reply by FaisalShaikhHA** [reply]
Thanks, @allenhwkim I am aware of this.
